### PR TITLE
Update JSON processing to use Thot.

### PR DESCRIPTION
**Pull request by jgrund**
Use thot instead of `Fable.Powerpack.JSON` parsing.

Thot is a more stringent approach (inspired by Elm),
and will give a structured way to parse our data.

While we're at it, refactor the eventtypes to move into submodules
and re-org to only use the fields we need from incoming messages.

Also update the ZfsTypes to live in submodules.

Signed-off-by: Joe Grund <joe.grund@intel.com>

### CODE REVIEW

File: IML/DeviceScannerDaemon/Handlers.fs [hunk 1]
```diff
@@ -5,111 +5,69 @@
...
+      ``end`` None
+    | Zpool.Create x ->
+      data.zpools <- Map.add x.guid x data.zpools
+      ``end`` None
+    | Zpool.Import x | Zpool.Export x ->
+      data.zpools <- Map.add x.guid x data.zpools
+      ``end`` None
+    | Zpool.Destroy x ->
+      data.zpools <- Map.remove x.guid data.zpools
+      // Remove any datasets and props.
```
Comment (tanabarr) [inline IML/DeviceScannerDaemon/Handlers.fs]: is this a "todo"?

File: IML/DeviceScannerDaemon/TestFixtures.fs [hunk 2]
```diff
@@ -86,7 +136,7 @@ let addMdraidObj = toMap """
 }
 """
 
-let removeObj = toMap """
+let removeJson = toJson """
```
Comment (tanabarr) [inline IML/DeviceScannerDaemon/TestFixtures.fs]: we should probably rename this so as to be consistent with the below definitions, for example to "removeSerialised" or "removeMessage" (and then append the relevant suffix to the below definitions to be consistent)

Comment (jgrund) [reply IML/DeviceScannerDaemon/TestFixtures.fs]: yes, meant to do that, I'll fix.

File: IML/DeviceScannerDaemon/Handlers.fs (hunk 1 above)
Comment (jgrund) [reply IML/DeviceScannerDaemon/Handlers.fs]: It's a todo for this patch. This isn't done, just pushed it early to be advisory of changes.

**Pull request merged**